src/js/components/widget.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { Header } from 'components/header.jsx';
+import { Conversation } from 'components/conversation.jsx';
+import { Settings } from 'components/settings.jsx';
+import { Notification } from 'components/notification.jsx';
+import { ChatInput } from 'components/chat-input.jsx';
+
+export class WidgetComponent extends Component {
+
+    render() {
+        const mainComponent = this.props.appState.settingsVisible ? <Settings /> : <Conversation />;
+        const footer = this.props.appState.settingsVisible ? null : <ChatInput />;
+
+        let className = typeof this.props.appState.widgetOpened === 'undefined' ? '' :
```

Comment:
That's a good question and it's mainly due to the animation to open and close the widget. `undefined` means it's in its default state, was never opened, was never closed. Once opened, the widget add CSS classes to manage the animations to open and close.

I should probably put that explanation in a comment on top of that line.